src/plugins/embeddable/public/plugin.tsx

```diff
@@ -52,90 +30,19 @@ import { getAllMigrations } from '../common/lib/get_all_migrations';
+} from './types';
 
-export interface EmbeddableSetupDependencies {
-  uiActions: UiActionsSetup;
-}
-
-export interface EmbeddableStartDependencies {
-  uiActions: UiActionsStart;
-  inspector: InspectorStart;
-  usageCollection: UsageCollectionStart;
-  contentManagement: ContentManagementPublicStart;
-  savedObjectsManagement: SavedObjectsManagementPluginStart;
-  savedObjectsTaggingOss?: SavedObjectTaggingOssPluginStart;
-}
-
-export interface EmbeddableSetup {
-  /**
-   * Register a saved object type with the "Add from library" flyout.
-   *
-   * @example
-   *  registerAddFromLibraryType({
-   *    onAdd: (container, savedObject) => {
-   *      container.addNewPanel({
-   *        panelType: CONTENT_ID,
-   *        initialState: savedObject.attributes,
-   *      });
-   *    },
-   *    savedObjectType: MAP_SAVED_OBJECT_TYPE,
-   *    savedObjectName: i18n.translate('xpack.maps.mapSavedObjectLabel', {
-   *      defaultMessage: 'Map',
-   *    }),
-   *    getIconForSavedObject: () => APP_ICON,
-   *  });
-   */
-  registerAddFromLibraryType: typeof registerAddFromLibraryType;
-
-  /**
-   * Registers an async {@link ReactEmbeddableFactory} getter.
-   */
-  registerReactEmbeddableFactory: typeof registerReactEmbeddableFactory;
-
-  /**
-   * @deprecated use {@link registerReactEmbeddableFactory} instead.
-   */
-  registerEmbeddableFactory: <
-    I extends EmbeddableInput,
-    O extends EmbeddableOutput,
-    E extends IEmbeddable<I, O> = IEmbeddable<I, O>
-  >(
-    id: string,
-    factory: EmbeddableFactoryDefinition<I, O, E>
-  ) => () => EmbeddableFactory<I, O, E>;
-  /**
-   * @deprecated
-   */
-  registerEnhancement: (enhancement: EnhancementRegistryDefinition) => void;
-}
-
-export interface EmbeddableStart extends PersistableStateService<EmbeddableStateWithType> {
-  /**
-   * @deprecated use {@link registerReactEmbeddableFactory} instead.
-   */
-  getEmbeddableFactory: <
-    I extends EmbeddableInput = EmbeddableInput,
-    O extends EmbeddableOutput = EmbeddableOutput,
-    E extends IEmbeddable<I, O> = IEmbeddable<I, O>
-  >(
-    embeddableFactoryId: string
-  ) => EmbeddableFactory<I, O, E> | undefined;
-
-  /**
-   * @deprecated
-   */
-  getEmbeddableFactories: () => IterableIterator<EmbeddableFactory>;
-  getStateTransfer: (storage?: Storage) => EmbeddableStateTransfer;
-}
 export class EmbeddablePublicPlugin implements Plugin<EmbeddableSetup, EmbeddableStart> {
-  private readonly embeddableFactoryDefinitions: Map<string, EmbeddableFactoryDefinition> =
-    new Map();
-  private readonly embeddableFactories: EmbeddableFactoryRegistry = new Map();
-  private readonly enhancements: EnhancementsRegistry = new Map();
   private stateTransferService: EmbeddableStateTransfer = {} as EmbeddableStateTransfer;
-  private isRegistryReady = false;
   private appList?: ReadonlyMap<string, PublicAppInfo>;
   private appListSubscription?: Subscription;
+  private enhancementsRegistry = new EnhancementsRegistry();
```

Comment:
Good call to move the registry code into its own file rather than clogging up the plugin.